frontend/src/components/Dashboard/CrossingListPage/CrossingListItem/CrossingListItem.js

```diff
@@ -31,6 +32,98 @@ class CrossingListItem extends React.Component {
     };
   }
 
+  componentWillMount() {
+    const { restoreDirtyState, crossing } = this.props;
+    const savedState = restoreDirtyState(crossing.id);
+    if (savedState) {
+      this.setState({ selectedStatus: savedState.selectedStatus });
+      this.setState({ selectedReason: savedState.selectedReason });
+      this.setState({ selectedDuration: savedState.selectedDuration });
+      this.setState({ notes: savedState.notes });
+      this.props.clearMeasurerCache();
+    }
+  }
+
+  componentWillUnmount() {
+    const { saveDirtyState } = this.props;
+
+    const stateToSave = {
+      crossingId: this.props.crossing.id,
+      selectedStatus: this.state.selectedStatus,
+      selectedDuration: this.state.selectedDuration,
+      selectedReason: this.state.selectedReason,
+      notes: this.state.notes
+    };
+
+    if (this.isDirty()) {
+      saveDirtyState(stateToSave);  
+    }
+  }
+
+  fixSort(store, updatedCrossing) {
+    // Get the edge from the current query
+    const { crossingQueryVariables } = this.props;
+    const data = store.readQuery({ query: crossingsQuery, variables: crossingQueryVariables });
+    const edge = data.searchCrossings.edges.find(edge => edge.node.id == updatedCrossing.id);
+
+    // Get all the query variable combinations we have cached
+    const queryVariables = Object.keys(store.data.ROOT_QUERY)
+                            .filter(query => query.includes('searchCrossings'))
+                            .map(q => JSON.parse(q.replace(/.*\(|\)/g, '')));
```

Comment:
Is this regex removing parens?

Reply:
Yes, also it's replacing the query name, so we are just parsing the variables object

<img width="1479" alt="screen shot 2017-12-08 at 12 50 00 pm" src="https://user-images.githubusercontent.com/6384156/33780460-72adb418-dc16-11e7-9b2e-633ea0188e6c.png">

Reply:
<img width="1475" alt="screen shot 2017-12-08 at 12 59 14 pm" src="https://user-images.githubusercontent.com/6384156/33780787-a4393fba-dc17-11e7-9e75-efe67fe5f66b.png">

ooh! I see the problem, good catch!

Reply:
<img width="1481" alt="screen shot 2017-12-08 at 1 07 13 pm" src="https://user-images.githubusercontent.com/6384156/33781062-c04556f2-dc18-11e7-88ec-1a134c5815f1.png">

Fixed it!